### PR TITLE
Level 1 implementation

### PR DESCRIPTION
**Pull request by kbence**
Work in progress

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module github.com/kbence/conndetect
+
+go 1.13
+
+require (
+	github.com/spf13/cobra v1.1.3
+	k8s.io/utils v0.0.0-20210305010621-2afb4311ab10
```
Comment (wadells) [inline go.mod]: This doesn't appear to be used.  `go mod tidy`?

File: internal/connlib/tcp.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+var fieldSeparator = regexp.MustCompile("\\s+")
+
+// ParseEndpoint - Parses and hexadecimal IPv4 endpoint (eg.: "0100007F:1F90")
+func ParseEndpoint(endpoint string) Endpoint {
+	parts := strings.Split(endpoint, ":")
+
+	ip, err := strconv.ParseUint(parts[0], 16, 32)
+	if err != nil {
+		panic(err)
```
Comment (wadells) [inline internal/connlib/tcp.go]: Panic is atypical chose for a go library function.  Why this over propagating the error upwards?

**Review by wadells — APPROVED** [review]
Looks great!  Will look out for directionality later per your remark.

File: go.mod (hunk 1 above)
Comment (kbence) [reply go.mod]: Ah, VSCode is trying to mess with me :D. It had accidentally included a k8s import. I didn't notice that it has also added it to `go.mod`. Thanks!

File: internal/connlib/tcp.go (hunk 2 above)
Comment (kbence) [reply internal/connlib/tcp.go]: Oh yeah, that was a quick and dirty one missed. Thank you!

File: test.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+ROOT=$(cd "$(dirname ${BASH_SOURCE[0]:-$0})"; pwd)
+
+cd "$ROOT"
+go test $(find . -name '*_test.go' | sed 's/[^/]\+$//' | uniq)
```
Comment (wadells) [inline test.sh:6]: What advantage does this `$(find ...)` provide over `./...`?

File: internal/connlib/tcp.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+			return DirectionalConnection{Source: conn.Remote, Destination: conn.Local}
+		}
+	}
+
+	// Check if there's a listener on 0.0.0.0 for the same port
+	// Note: this is totally not right, since we don't have any
+	// information about what IP addresses this specific node has.
+
+	// TODO: parse it out from /proc/net/fib_trie and use that
+	// list instead of IP.IsUnspecified()!
```
Comment (wadells) [inline internal/connlib/tcp.go:122]: Thanks for these comments.  Even though the work is out of scope, it is nice to know how you'd approach the next step.

**Review by wadells — APPROVED** [review]
New set of patches also looks good to me!  Nice testing.

File: test.sh (hunk 3 above)
Comment (kbence) [reply test.sh:6]: I figured later on that `go test` can only run one package at the time, so I put it in a `for` loop, but I just expected to work if `*_test.go` files are popping up elsewhere. I was naive :smile:

**Pull request merged**